### ISSUE TITLE
[Feature Request] Privacy-safe community telemetry export and collaborative HyDRA model calibration loop

### ISSUE THREAD

**Issue by beettlle** [beettlle/pi-smart-router#36]
# Objective
Implement a privacy-safe community telemetry export utility and an automated build-time pipeline to collaboratively calibrate the **HyDRA matcher's linear projection weights (`384 × 3`)** and triage thresholds with every project release.

This enables a **federated, privacy-preserving feedback loop**—where the community's collective routing experiences (e.g., user overrides, model accuracy, and latencies) are mathematically compiled and distributed to improve model-routing accuracy for all users.

---

# Why It's Needed & The Technical Blueprint

## 1. Zero-Knowledge, Privacy-Safe Data Foundation
The project's SQLite storage is already architected to be privacy-safe. The `dataset` table (defined in `src/infrastructure/persistence/sqlite-store.ts`) stores **zero raw prompt text or proprietary code**. It only records:
*   Physical prompt metrics (`prompt_length_chars`, `estimated_input_tokens`)
*   Control-flow features (`triage_cyclomatic_score`, keyword hits)
*   Capability weights (`requirement_reasoning`, `requirement_code_gen`, `requirement_tool_use`)
*   Cryptographic hashes for repeat detection (`prompt_fingerprint`)
*   Result outcomes (routing stage, latency, estimated cost, and whether a user triggered a manual model override)

Because the database contains absolutely no sensitive IP, users can safely share these telemetry snapshots without risking data or credential leakage.

## 2. Closing the Loop on the HyDRA Matcher
Currently, the projection from a 384-dimensional ONNX prompt embedding to the 3 requirement dimensions (Reasoning, Code Gen, Tool Use) inside `src/domain/matching/hydra-matcher.ts` uses a mean-pooled slice-and-sigmoid placeholder. The comments state:
```typescript
/**
 * Fixed deterministic projection from 384-dim embedding to 3 requirement
 * dimensions via mean-pooled thirds + sigmoid normalization.
 *
 * Production refinement: replace with a learned linear projection loaded
 * from artifact cache (Phase 2).
 */
export function projectToRequirements(embedding: Float32Array): RequirementVector { ... }
```

By gathering anonymized community datasets linking prompt features to user override behaviors, we can train a **learned projection head (`384 × 3` weights)** that is optimized against real-world developer inputs.

---

# Proposed Feature Architecture

We propose introducing this capability in three segments:

### A. The CLI Exporter (`telemetry:export`)
Add a command to export the local `dataset` and `outcomes` SQLite records to a standard, anonymized JSON payload:
```bash
pi-smart-router telemetry:export --output=telemetry_contrib.json
```
This script will read the local database, strip any human-readable fields, and compile the feature vectors and decision metrics into a standardized format suitable for sharing.

### B. The Community Feedback Submission
Establish a designated submission channel:
*   A pinned **GitHub Discussion** or **GitHub Issue Template** where users can copy-paste their anonymized JSON payload.
*   Alternatively, an automated contribution script that submits the JSON payload directly to a centralized server or opens a PR against a `data/contrib/` directory.

### C. Build-Time Training & Platform-Agnostic Redistribution
Within the project’s CI/CD or release pipeline:
1.  **Aggregate:** Consolidate all user-submitted JSON telemetry logs.
2.  **Train:** Run a lightweight linear regression or single-layer perceptron training script (using scikit-learn, NumPy, or PyTorch) to find the optimal weights that minimize user override outcomes and maximize routing efficiency.
3.  **Serialize:** Compile the trained `384 × 3` matrix and optimal triage boundaries to an architecture-agnostic **`weights.json`** or raw binary float32 array file (e.g., `src/assets/projection_weights.json`, ~15 KB).
4.  **Package:** Distribute `weights.json` as a static asset inside the NPM/CLI package.
5.  **Execute:** Update `projectToRequirements()` inside `HydraMatcher` to load these weights on startup and execute a standard matrix multiplication (`embedding × weights`) on the user’s machine. Since this is simple floating-point math running in native Node.js/TypeScript, it runs with complete bit-for-bit parity on all architectures (macOS, Windows, Linux, x86, ARM64).

---

# Verification & Testability
*   Provide a utility script inside the project (`npm run telemetry:test-projection`) that runs a set of standard benchmarking prompts against the current compiled weights and prints expected routing decisions.
*   Ensures that updating the model projection weights does not introduce latency overhead beyond the current `80–120ms` matching budget.

**Comment by beettlle** [comment]
Landed via **SP-082**: privacy-safe community telemetry export in `src/infra/telemetry.ts` with unit tests at `tests/unit/telemetry-export.test.ts`.

Verification: `npm run typecheck && npm test` pass on main.
